lib/serialization/Serializer.js

```diff
@@ -15,7 +15,11 @@ class Serializer {
 		const ctx = { ...context, ...this.context };
 		let current = obj;
 		for (const middleware of this.serializeMiddlewares) {
-			if (current instanceof Promise) {
+			if (
+				!!current &&
+				(typeof current === "object" || typeof current === "function") &&
```

Comment:
I wonder why are you testing for current being a function? Does promises exist that are functions?

Reply:
yes, Because it says in the specification: "[promise” is an object or function with a then method](https://promisesaplus.com/)"